TestStack.Dossier.Tests/BuilderBuildTests.cs

```diff
@@ -0,0 +1,56 @@
+﻿using Shouldly;
+using TestStack.Dossier.Tests.Builders;
+using TestStack.Dossier.Tests.Entities;
+using Xunit;
+
+namespace TestStack.Dossier.Tests
+{
+    public class BuilderBuildTests
+    {
+        [Fact]
+        public void GivenBasicBuilder_WhenCallingBuildExplicitly_ThenReturnAnObject()
+        {
+            var builder = Builder<Customer>.CreateNew();
+
+            var customer = builder.Build();
+
+            customer.ShouldBeOfType<Customer>();
+        }
+
+        [Fact]
+        public void GivenBuilder_WhenCallingSetExplicitly_ShouldOverrideValues()
+        {
+            var builder = Builder<Customer>.CreateNew()
```

Comment:
But if we have a default implementation of BuildObject it will get used by people for DDD objects because it won't be clear that 's not a good idea (it wouldn't necessarily be obvious what was happening).

I wonder if for now we put this BuildObject definition in `Builder<T>` for now to avoid these issues for this PR then we can look at how we support a default implementation of BuildObject as part of addressing #24?

It's still slightly worrying to me that it sets private setter values though - I'm not sure if that would be expected behaviour. I wouldn't object if there was a config option to turn that on (but I think it should be opt in rather than opt out).